src/transformStateWithClones.js

```diff
@@ -7,7 +7,29 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  const copy = { ...state };
+
+  for (const action in actions) {
+    if (actions[action].type === 'addProperties') {
+      Object.assign(copy, actions[action].extraData);
+    };
```

Comment:
В переменной action на каждой итерации хранится объект из массива actions, к нему можно обращаться напрямую, без []:
```suggestion
  for (const action in actions) {
    if (action.type === 'addProperties') {
      Object.assign(copy, action.extraData);
    };
```